components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/AccountLockHandler.java

```diff
@@ -193,33 +195,40 @@ protected boolean handlePreAuthentication(Event event, String userName, UserStor
             IdentityUtil.setIdentityErrorMsg(customErrorMessageContext);
         }
 
-        if (isAccountLock(userName, userStoreManager)) {
-            //User account is locked. If the current time is not exceeded user unlock time, send a error message
-            // saying user is locked, otherwise users can try to authenticate and unlock their account upon a
-            // successful authentication.
+        if (!isAccountLockBypassRoleExist(userStoreManager, userName)) {
```

Comment:
We can optimize the logic here, no need to check isAccountLockBypassRoleExist, when account is not locked